tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -66,6 +69,20 @@ final class ScalaJSClassEmitter private (
     linkingUnit.classDefs.map(c => c.encodedName -> c).toMap
   }
 
+  private[javascript] lazy val hasSubclasses: Set[String] = {
+    if (linkingUnit == null) {
+      throw new IllegalArgumentException(
+          "A class emitter created without a LinkingUnit cannot emit JS classes")
+    }
+    linkingUnit.classDefs.map(c => c.superClass).
+      collect{case Some(x) => x.name}.toSet
+  }
+
+  def effectivelyFinal(clss: LinkedClass): Boolean = {
+    !hasSubclasses(clss.name.name) &&
+    clss.classExports.isEmpty
```

Comment:
You're right, I should have changed the name when adding the condition to filter out classes with export as to not optimize them.